### PR TITLE
⚠️ Remove deprecated manager options

### PR DESCRIPTION
**Pull request by troy0820**
`options.AndFrom` and `options.AndFromOrDie` have been deprecated over several releases. This PR will remove them where consumers will have to migrate to their own config solution.   

- [x] Remove function `options.AndFrom`
- [x] Remove function `options.AndFromOrDie`

/kind cleanup

### CODE REVIEW

**Review by camilamacedo86 — COMMENTED** [review]
This code ensures that the options work
So, IMO it should only be removed when the options be removed and are no longer supported.

**Comment by inteon** [comment]
I think we can only remove these functions when we remove `sigs.k8s.io/controller-runtime/pkg/config` too.

**Comment by troy0820** [comment]
> This code ensures that the options work
So, IMO it should only be removed when the options be removed and are no longer supported.

The `options.AndFrom` was loading a config that @inteon agreed that we can only deprecate this if we deprecate the config package.  

Do you mean that the options that get merged with the config options shouldn't be removed? This doesn't remove any options.

https://github.com/kubernetes-sigs/controller-runtime/issues/895

> 👍 I think we can only remove these functions when we remove sigs.k8s.io/controller-runtime/pkg/config too.

Very true, I can delete those as well.

File: Makefile [hunk 1]
```diff
@@ -102,10 +102,6 @@ modules: ## Runs go mod to ensure modules are up to date.
 	cd $(ENVTEST_DIR); go mod tidy
 	cd $(SCRATCH_ENV_DIR); go mod tidy
 
-.PHONY: generate
-generate: $(CONTROLLER_GEN) ## Runs controller-gen for internal types for config file
-	$(CONTROLLER_GEN) object paths="./pkg/config/v1alpha1/...;./examples/configfile/custom/v1alpha1/..."
```
Comment (camilamacedo86) [inline Makefile:107]: Why it has been removed?

File: Makefile [hunk 2]
```diff
@@ -126,9 +122,4 @@ verify-modules: modules ## Verify go modules are up to date
 		echo "go module files are out of date, please run 'make modules'"; exit 1; \
 	fi
 
-.PHONY: verify-generate
-verify-generate: generate ## Verify generated files are up to date
-	@if !(git diff --quiet HEAD); then \
-		git diff; \
-		echo "generated files are out of date, run make generate"; exit 1; \
-	fi
```
Comment (camilamacedo86) [inline Makefile:134]: Why it has been removed?

**Comment by troy0820** [comment]
> /lgtm PR LGTM, we now just need an approver to agree that we want to remove this deprecated feature already.

Thank you for the review, I know that this has to be coordinated and maybe targeted for v0.18.0.

File: Makefile [hunk 3]
```diff
@@ -116,10 +116,6 @@ modules: ## Runs go mod to ensure modules are up to date.
 	cd $(ENVTEST_DIR); go mod tidy
 	cd $(SCRATCH_ENV_DIR); go mod tidy
```
Comment (sbueringer) [inline Makefile:118]: verify-generate can be deleted as well

Comment (troy0820) [reply Makefile:118]: I will make this change.

**Comment by sbueringer** [comment]
Thank you very much!

/lgtm

/assign @alvaroaleman @vincepri

**Comment by alvaroaleman** [comment]
/retitle ⚠️ Remove deprecated manager options

The current title sounds to me like we were un-deprecating something, WDYT?
/lgtm
/approve
/hold

In case @vincepri wants to have a look

**Comment by sbueringer** [comment]
@troy0820 Thank you very much!

**Pull request merged**